src/coreComponents/constitutive/fluid/multifluid/CO2Brine/functions/CO2EOSSolver.cpp

```diff
@@ -61,14 +61,7 @@ CO2EOSSolver::solve( string const & name,
     if( var < allowedMinValue )
     {
       var = allowedMinValue;
-      (*f)( temp, pres, var );
-    }
-
-    // check convergence (based on the magnitude of the Newton update for historical reasons)
-    if( LvArray::math::abs( update ) < tolerance )
-    {
-      newtonHasConverged = true;
-      break;
+      res = (*f)( temp, pres, var );
```

Comment:
this seems to also help to fix the spikes, independently from residual convergence check